src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -11135,14 +11598,15 @@ public void OrderOfEvaluation()
                 using System;
                 using System.Collections;
                 using System.Collections.Generic;
-                class C<T> : IEnumerable
+                class C<T> : IEnumerable<T>
                 {
                     private List<T> _list = new List<T>();
                     public void Add(T t)
                     {
                         Console.WriteLine("Add {0}", t);
                         _list.Add(t);
                     }
+                    IEnumerator<T> IEnumerable<T>.GetEnumerator() => _list.GetEnumerator();
```

Comment:
consider making the private `IEnumerable.GetEnumerator()` implementation below throw